src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -192,7 +219,10 @@ export class ExamExerciseImportComponent implements OnInit {
     validateTitleOfProgrammingExercise(exercise: Exercise): boolean {
         return (
             // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
-            exercise.title?.length! > 0 && this.titleNamePattern.test(exercise.title!) && exercise.title !== this.getBlocklistTitleOfProgrammingExercise(exercise.id!)
+            exercise.title?.length! > 0 &&
```

Comment:
With this you can remove the disable-next-line option above.
```suggestion
            exercise.title?.length &&
```

Reply:
```suggestion
            exercise.title!.length! > 0 &&
```
I think it's a bit weird to first use `?` and then `!` in a chain, because then it is only implicitly (not explicitly) assumed that the `title` is defined.

Reply:
If I change it to `exercise.title?.length` the code does not compile due to `Returned expression type 0 | undefined | false | boolean is not assignable to type boolean`. In case I change it to `exercise.title!.length! > 0` some tests, that test the method with `undefined` title fail. Since I did not write this part of the code, I would just leave it as it is now, or do you still think I should change it?

Reply:
Here's an alternative suggestion that should work:
```suggestion
            Boolean(exercise.title) &&
            exercise.title!.length > 0 &&
```
I think it's a red flag that the code assumes a defined title but works only because of `undefined > 0 === false` in the end, which seems a bit "lucky." 

If this still does not work because of some circumstance, I'm also fine with you leaving it as-is because you didn't introduce that aspect of the code, and it already was proven to work in production.

Reply:
You can fix the error by converting the falsy-value into a boolean:
```suggestion
            !!exercise.title?.length &&
```

Checking the length is needed here since otherwise an empty string would return true.

Reply:
This seems to work well, implemented :)